File: Driver/Safari/Selenium_with_Node_js/contextclick.js
const { Timeouts } = require('selenium-webdriver/lib/capabilities');
var assrt=require('assert');
const { assert } = require('console');
const { resolve } = require('path');
const { rejects } = require('assert');
var webdriver = require('selenium-webdriver'),
By=webdriver.By,
JavaScriptExecutor = webdriver.JavaScriptExecutor,
until = webdriver.until;

var driver = new webdriver.Builder().forBrowser('safari').build();

driver.get("https://swisnl.github.io/jQuery-contextMenu/demo.html").then(()=>{
    



    driver.actions().contextClick(driver.findElement(By.xpath("//*[text()='right click me']"))).perform()
    
})

File: Driver/Safari/My_Python_code/Context_click.py
from selenium import webdriver
import time
from selenium.webdriver.common import alert
from selenium.webdriver.support.select import Select
from selenium.webdriver.common.alert import Alert
from selenium.webdriver.safari.service import Service
from selenium.webdriver.common.action_chains import ActionChains



driver  = webdriver.Safari(executable_path='/usr/bin/safaridriver')

driver.maximize_window()

driver.get("https://swisnl.github.io/jQuery-contextMenu/demo.html")

ele=driver.find_element_by_xpath("//*[text()='right click me']")



action = ActionChains(driver)
  
# context click the item
action.context_click(on_element=ele)
action.perform()
print("right click success")
time.sleep(5)
driver.quit()